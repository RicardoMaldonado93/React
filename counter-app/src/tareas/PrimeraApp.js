
import React from 'react';
import PropTypes from 'prop-types'


/** FC */
const PrimeraApp = ( { saludo, subtitulo } )=>{
    // const saludo = "Hola mundo!";
    const jsonObject = {
        nombre: "Ricardo",
        edad: 27
    }

    
    return (
        <>
            <h1>{ saludo }</h1>
            {/* <pre>{ JSON.stringify( jsonObject, null, 3 ) }</pre> */}
            {/* <p>Mi primera aplicacion</p> */}
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo : PropTypes.string
}

PrimeraApp.defaultProps = {
    subtitulo : "soy un subtitulo"
}


 export default PrimeraApp;