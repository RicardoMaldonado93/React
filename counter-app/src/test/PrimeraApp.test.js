// import { render } from "@testing-library/react";
// import PrimeraApp from "./PrimeraApp";
// import React from 'react';
import React from 'react';
import { shallow } from 'enzyme';
import "@testing-library/jest-dom";
import PrimeraApp from '../tareas/PrimeraApp';

// describe("Pruebas con el componente 'PrimeraApp'", () => {
    
//     test("Debe de mostrar el mensaje 'Hola soy Goku'", () => {
        
//         const saludo = "Hola soy Goku";
//         const { getByText } = render(<PrimeraApp saludo={ saludo }/>);

//         expect( getByText(saludo) ).toBeInTheDocument()
//     });
// })


describe("Pruebas componente '<PrimeraApp />'",()=>{

    test("debe mostrar '<PrimeraApp />' correctamente", ()=>{

        const saludo = "Hola soy Goku";
        const wrapper = shallow( <PrimeraApp saludo={saludo} /> )

        expect( wrapper ).toMatchSnapshot();
    })
})