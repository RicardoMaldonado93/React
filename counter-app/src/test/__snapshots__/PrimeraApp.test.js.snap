// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente '<PrimeraApp />' debe mostrar '<PrimeraApp />' correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
